### PR TITLE
Added warnings for bad presence update payloads.

### PR DESCRIPTION
**Pull request by nekokatt**
This makes some warnings appear if you send deprecated or badly formatted presence updates.

Should stop people asking "why does this not work" :)

### CODE REVIEW

File: hikari/impl/bot.py [hunk 1]
```diff
@@ -990,3 +995,34 @@ async def murder(future: asyncio.Future[typing.Any]) -> None:
 
         _LOGGER.debug("closing event loop")
         loop.close()
+
+    @staticmethod
+    def _validate_activity(activity: undefined.UndefinedNoneOr[presences.Activity]) -> None:
```
Comment (FasterSpeeding) [inline hikari/impl/bot.py:1000]: does this have to be on the class?

File: hikari/impl/bot.py [hunk 2]
```diff
@@ -990,3 +995,34 @@ async def murder(future: asyncio.Future[typing.Any]) -> None:
...
+        # This seems to cause confusion for a lot of people, so lets add some warnings into the mix.
+
+        if activity is undefined.UNDEFINED or activity is None:
+            return
+
+        # If you ever change where this is called from, make sure to check the stacklevels are correct
+        # or the code preview in the warning will be wrong...
+        if activity.type == presences.ActivityType.WATCHING:
```
Comment (FasterSpeeding) [inline hikari/impl/bot.py:1008]: could this not be `is` instead of `==`?

File: hikari/impl/bot.py (hunk 1 above)
Comment (nekokatt) [reply hikari/impl/bot.py:1000]: yes, the stacklevel has to be hardcoded in the warnings to make the line of code it shows in the warning make any sense.

File: hikari/impl/bot.py [hunk 3]
```diff
@@ -990,3 +995,34 @@ async def murder(future: asyncio.Future[typing.Any]) -> None:
...
+            warnings.warn(
+                "The WATCHING activity type is not officially recognised by Discord and may be removed from Hikari in "
+                "a future release.",
+                category=PendingDeprecationWarning,
+                stacklevel=3,
+            )
+        elif activity.type == presences.ActivityType.CUSTOM:
```
Comment (FasterSpeeding) [inline hikari/impl/bot.py:1015]:
```suggestion
        elif activity.type is presences.ActivityType.CUSTOM:
```

File: hikari/impl/bot.py [hunk 4]
```diff
@@ -990,3 +995,34 @@ async def murder(future: asyncio.Future[typing.Any]) -> None:
...
+            warnings.warn(
+                "The CUSTOM activity type is not supported by bots at the time of writing, and may therefore not have "
+                "any effect if used.",
+                category=errors.HikariWarning,
+                stacklevel=3,
+            )
+        elif activity.type == presences.ActivityType.STREAMING and activity.url is None:
```
Comment (FasterSpeeding) [inline hikari/impl/bot.py:1022]:
```suggestion
        elif activity.type is presences.ActivityType.STREAMING and activity.url is None:
```

**Review by FasterSpeeding — APPROVED** [review]
lgtm

**Review by davfsa — APPROVED** [review]

**Pull request merged**